src/robot.cpp

```diff
@@ -101,8 +101,7 @@ Robot::loadFromXml(const TiXmlElement* root,QString rootPath)
 		return FAILURE;
 	}
 	QString robotDBName = root->Attribute("DBName");
-	if(robotDBName.isNull()) robotDBName = robotName;
-        setDBName(robotDBName);
+	setDBName(robotDBName);
```

Comment:
Feels like the robot's name is a better default than Null (or whatever root->Attribute returns if the attribute is not found)?